File: examples/test-projects/small-projects/test-project-java/FileStorage.java
package testproject;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

/** Файловое хранилище — аналог C++ file_storage. Данные циркулируют через файлы. */
public class FileStorage {

    // Сохраняет значение счётчика в файл (запись в файл — ИО-файл)
    public void saveCounter(int value) throws IOException {
        BufferedWriter writer = new BufferedWriter(new FileWriter("counter.dat"));
        writer.write(Integer.toString(value));
        writer.close();
    }

    // Загружает значение счётчика из файла (чтение из файла)
    public int loadCounter() throws IOException {
        BufferedReader reader = new BufferedReader(new FileReader("counter.dat"));
        int value = Integer.parseInt(reader.readLine());
        reader.close();
        return value;
    }

    // Дописывает строку в лог-файл (запись в файл)
    public void appendLog(String message) throws IOException {
        BufferedWriter writer = new BufferedWriter(new FileWriter("app.log", true));
        writer.write(message);
        writer.newLine();
        writer.close();
    }

    // Читает первую строку из лог-файла (чтение из файла)
    public String readLog() throws IOException {
        BufferedReader reader = new BufferedReader(new FileReader("app.log"));
        String line = reader.readLine();
        reader.close();
        return line;
    }
}

File: examples/test-projects/small-projects/test-project-python/file_storage.py
"""Файловое хранилище — аналог FileStorage из других малых проектов."""


class FileStorage:
    """Хранит счётчик и лог в текстовых файлах."""

    def save_counter(self, value):
        f = open("counter.dat", "w")
        f.write(str(value))
        f.close()

    def load_counter(self):
        f = open("counter.dat", "r")
        value = int(f.read())
        f.close()
        return value

    def append_log(self, message):
        f = open("app.log", "a")
        f.write(message + "\n")
        f.close()

    def read_log(self):
        f = open("app.log", "r")
        line = f.readline()
        f.close()
        return line
